File: dmn-core/src/test/resources/tck/cl2/expected/dmn/0004-simpletable-u/ApprovalStatusRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "ApprovalStatus"})
public class ApprovalStatusRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private String approvalStatus;

    public ApprovalStatusRuleOutput(boolean matched) {
        super(matched);
    }

    public String getApprovalStatus() {
        return this.approvalStatus;
    }
    public void setApprovalStatus(String approvalStatus) {
        this.approvalStatus = approvalStatus;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ApprovalStatusRuleOutput other = (ApprovalStatusRuleOutput) o;
        if (this.getApprovalStatus() != null ? !this.getApprovalStatus().equals(other.getApprovalStatus()) : other.getApprovalStatus() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getApprovalStatus() != null ? this.getApprovalStatus().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", approvalStatus='%s'", approvalStatus));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-test-cases/tck/1.1/cl2/0004-simpletable-U/translator/expected/kotlin/dmn/ApprovalStatusRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "ApprovalStatus"])
class ApprovalStatusRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("ApprovalStatus")
    var approvalStatus: String? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as ApprovalStatusRuleOutput
        if (if (this.approvalStatus != null) this.approvalStatus != other.approvalStatus else other.approvalStatus != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.approvalStatus != null) this.approvalStatus.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", approvalStatus='%s'", approvalStatus))
        result_.append(")")
        return result_.toString()
    }
}
